src/gui/qgsmapcanvas.cpp

```diff
@@ -147,6 +147,8 @@ QgsMapCanvas::QgsMapCanvas( QWidget *parent )
   mSettings.setFlag( QgsMapSettings::DrawEditingInfo );
   mSettings.setFlag( QgsMapSettings::UseRenderingOptimization );
   mSettings.setFlag( QgsMapSettings::RenderPartialOutput );
+  mSettings.setDestinationCrs( QgsProject::instance()->crs() );
+  mSettings.setEllipsoid( QgsProject::instance()->ellipsoid() );
```

Comment:
This should do it.
<pre>
  connect( QgsProject::instance(), &QgsProject::crsChanged,
           this, [ = ]
  {
    mSettings.setEllipsoid( QgsProject::instance()->ellipsoid() );
  } );
</pre>